utils/profile-services.ts

```diff
@@ -0,0 +1,217 @@
+                ResponseErrorFromValidationErrors(RetrievedServicePreference)
+              )
+            ),
+          e.left(missingServicePreference),
+          (h, _t) =>
+            h.mapLeft(_errors =>
+              ResponseErrorNotFound(
+                NOT_FOUND_TITLE,
+                "Can not decode the RetrievedServicePreference"
+              )
+            )
+        )
+      )
+    );
+
+const servicePreferenceToLimitedProfile = (
+  profile: RetrievedProfile,
+  servicePreference: RetrievedServicePreference
+): LimitedProfile =>
+  retrievedProfileToLimitedProfile(profile, servicePreference.isInboxEnabled);
+
+export const profileWithPreferenceVersionWithModeAuto: IServicePreferenceHandler = {
+  handleProfile: (profile, servicesPreferencesModel, serviceId) =>
+    findServicePreference(servicesPreferencesModel, profile, serviceId).fold(
+      _l => retrievedProfileToLimitedProfile(profile, true),
+      servicePreference =>
+        servicePreferenceToLimitedProfile(profile, servicePreference)
+    ),
+  isMyReposability: profile =>
+    profile.servicePreferencesSettings.version >= 0 &&
+    profile.servicePreferencesSettings.mode === ServicesPreferencesModeEnum.AUTO
+};
+
+export const profileWithPreferenceVersionWithModeManual: IServicePreferenceHandler = {
+  handleProfile: (profile, servicesPreferencesModel, serviceId) =>
+    findServicePreference(servicesPreferencesModel, profile, serviceId).fold(
+      _l => retrievedProfileToLimitedProfile(profile, false),
+      servicePreference =>
+        servicePreferenceToLimitedProfile(profile, servicePreference)
+    ),
+  isMyReposability: profile =>
+    profile.servicePreferencesSettings.version >= 0 &&
+    profile.servicePreferencesSettings.mode ===
+      ServicesPreferencesModeEnum.MANUAL
+};
+
+export const profileWithModeLegacy: IServicePreferenceHandler = {
+  handleProfile: (profile, _servicesPreferencesModel, serviceId) =>
+    t.task.of(
+      retrievedProfileToLimitedProfile(
+        profile,
+        isSenderAllowed(profile.blockedInboxOrChannels, serviceId)
+      )
+    ),
+  isMyReposability: profile =>
+    profile.servicePreferencesSettings.mode ===
+    ServicesPreferencesModeEnum.LEGACY
+};
+
+export const handle = (handler: IServicePreferenceHandler) => (
+  profile: RetrievedProfile,
+  servicesPreferencesModel: ServicesPreferencesModel,
+  serviceId: ServiceId
+): t.Task<e.Either<Error, LimitedProfile>> =>
+  te
+    .fromPredicate(
+      handler.isMyReposability,
+      _ => new Error("Handler not feasable!")
+    )(profile)
+    .chain(_ =>
+      te.right(
+        handler.handleProfile(profile, servicesPreferencesModel, serviceId)
+      )
+    )
+    .fold(
+      l => e.left(l),
+      r => e.right(r)
+    );
+
+export const handleAll = () => (
+  profile: RetrievedProfile,
+  servicesPreferencesModel: ServicesPreferencesModel,
+  serviceId: ServiceId
+): te.TaskEither<IResponseErrorNotFound, LimitedProfile> =>
```

Comment:
What does this function do? _handleAll_ what?

Beside of naming, which is subjective and debatable, try to explain the **why** with a comment

Reply:
I think you should report this here https://github.com/pagopa/io-functions-services/pull/136/files#diff-4e0957bac5b9ec17e7bb60a713b90d20a27ce51b0003541ac27fb476b911158b